.github/actions/init/action.yaml

```diff
@@ -0,0 +1,133 @@
+                  ~/boost_1_82_0/
+                  ~/.emsdk/
+                  ~/.llvm/
+              key: ${{ runner.os }}-emsdk-${{ hashFiles('package.json') }}
+              restore-keys: |
+                  ${{ runner.os }}-emsdk-
+
+        - name: Setup pip cache
+          uses: actions/cache@v4
+          if: ${{ inputs.skip_cache == 'false' }}
+          with:
+              path: |
+                  ~/.cache/pip
+                  ~/py_modules/
+              key: ${{ runner.os }}-pip-${{ hashFiles('**/setup.py') }}
+              restore-keys: |
+                  ${{ runner.os }}-pip-
+
+        - name: Setup cargo cache
+          uses: actions/cache@v4
+          if: ${{ inputs.skip_cache == 'false' }}
+          with:
+              key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+              path: |
+                  ~/.cargo/bin/
+                  ~/.cargo/registry/index/
+                  ~/.cargo/registry/cache/
+                  ~/.cargo/git/db/
+                  rust/perspective-viewer/target/
+              restore-keys: |
+                  ${{ runner.os }}-cargo-
+
+        - name: ccache
+          uses: hendrikmuhs/ccache-action@v1.2
+          if: ${{ inputs.skip_cache == 'false' }}
+          with:
+              key: ${{ github.job }}-${{ matrix.os }}
+
+        - name: Set up Python ${{ matrix.python-version }}
+          uses: actions/setup-python@v5
+          with:
+              python-version: ${{ matrix.python-version }}
+              cache: "pip"
+
+        - name: Use Node.js ${{ matrix.node-version }}
+          uses: actions/setup-node@v4
+          with:
+              node-version: ${{ matrix.node-version }}
+              cache: "pnpm"
+              cache-dependency-path: pnpm-lock.yaml
+
+        - name: Install latest nightly rust
+          uses: dtolnay/rust-toolchain@nightly
+          with:
+              toolchain: nightly-2024-05-07
+              targets: wasm32-unknown-unknown
+              components: rustfmt, clippy, rust-src
+
+        - name: Install LLVM 17
+          uses: KyleMayes/install-llvm-action@v2
+          with:
+              version: "17"
+              directory: "./.llvm"
+              cached: true
+
+        # JS
+        - name: Install JS dependencies
+          shell: bash
+          run: pnpm install
+
+        - name: Install venv
+          shell: bash
+          run:
+
+        - name: Activate virtualenv
+          shell: bash
+          run: |
+              python3 -m venv py_modules
+              . ./py_modules/bin/activate
+              echo PATH=$PATH >> $GITHUB_ENV
+              echo VIRTUAL_ENV=$VIRTUAL_ENV >> $GITHUB_ENV
+              pip3 install maturin
+
+        - name: Linux init steps
+          shell: bash
+          run: sudo node tools/perspective-scripts/install_tools.mjs
+          if: ${{ runner.os == 'Linux' }}
+
+        - name: Install CCache
+          shell: bash
+          run: sudo apt install -y ccache
```

Comment:
this step is handled by `hendrikmuhs/ccache-action`

Reply:
👍 

I've implemented this in #2659